cmd/prysmctl/testnet/generate_genesis.go

```diff
@@ -0,0 +1,246 @@
+		cmd.ChainConfigFileFlag,
+		depositJsonFileFlag,
+		configNameFlag,
+		numValidatorsFlag,
+		genesisTimeFlag,
+		outputSSZFlag,
+		outputYamlFlag,
+		outputJsonFlag,
+	},
+}
+
+var (
+	log                 = logrus.WithField("prefix", "genesis")
+	depositJsonFileFlag = &cli.StringFlag{
+		Name:  "deposit-json-file",
+		Value: "",
+		Usage: "Path to deposit_data.json file generated by the staking-deposit-cli tool for optionally specifying validators in genesis state",
+	}
+	configNameFlag = &cli.StringFlag{
+		Name:  "config-name",
+		Usage: "Config kind to be used for generating the genesis state. Default: mainnet. Options include mainnet, interop, minimal, prater, ropsten, sepolia. --chain-config-file will override this flag.",
+		Value: params.MainnetName,
+	}
+	numValidatorsFlag = &cli.Uint64Flag{
+		Name:  "num-validators",
+		Usage: "Number of validators to deterministically generate in the genesis state",
+
+		Value: 0,
+	}
+	genesisTimeFlag = &cli.Uint64Flag{
+		Name:  "genesis-time",
+		Usage: "Unix timestamp seconds used as the genesis time in the genesis state. If unset, defaults to now()",
+	}
+	outputSSZFlag = &cli.StringFlag{
+		Name:  "output-ssz",
+		Usage: "Output filename of the SSZ marshaling of the generated genesis state",
+		Value: "",
+	}
+	outputYamlFlag = &cli.StringFlag{
+		Name:  "output-yaml",
+		Usage: "Output filename of the YAML marshaling of the generated genesis state",
+		Value: "",
+	}
+	outputJsonFlag = &cli.StringFlag{
+		Name:  "output-json",
+		Usage: "Output filename of the JSON marshaling of the generated genesis state",
+		Value: "",
+	}
+)
+
+// Represents a json object of hex string and uint64 values for
+// validators on Ethereum. This file can be generated using the official eth2.0-deposit-cli.
+type depositDataJSON struct {
+	PubKey                string `json:"pubkey"`
+	Amount                uint64 `json:"amount"`
+	WithdrawalCredentials string `json:"withdrawal_credentials"`
+	DepositDataRoot       string `json:"deposit_data_root"`
+	Signature             string `json:"signature"`
+}
+
+func cliActionGenerateGenesisState(cliCtx *cli.Context) error {
+	if !cliCtx.IsSet(genesisTimeFlag.Name) {
+		log.Infof("No %s specified, defaulting to now()", genesisTimeFlag.Name)
+	}
+	hasOutputFlag := !cliCtx.IsSet(outputJsonFlag.Name) ||
+		!cliCtx.IsSet(outputYamlFlag.Name) ||
+		!cliCtx.IsSet(outputSSZFlag.Name)
+	if !hasOutputFlag {
+		return fmt.Errorf(
+			"no %s, %s, or %s flag specified. At least one is required",
+			outputJsonFlag.Name,
+			outputYamlFlag.Name,
+			outputSSZFlag.Name,
+		)
+	}
+	if err := setGlobalParams(cliCtx); err != nil {
+		return fmt.Errorf("could not set config params: %v", err)
+	}
+	genesisState, err := generateGenesis(cliCtx)
+	if err != nil {
+		return fmt.Errorf("could not generate genesis state: %v", err)
+	}
+	outputJson := cliCtx.String(outputJsonFlag.Name)
+	outputYaml := cliCtx.String(outputYamlFlag.Name)
+	outputSSZ := cliCtx.String(outputSSZFlag.Name)
```

Comment:
There is no `outputSSZ != ""` check - is this required? If yes, then it should be properly verified beforehand if the flag was set.